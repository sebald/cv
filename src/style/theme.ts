export const theme = {
  baseline: 12,
  htmlFontSize: 16,
};
