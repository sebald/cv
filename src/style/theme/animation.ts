import { keyframes } from '../styled-components';

export const fadeInUp = keyframes`
  from {
    opacity: 0;
    transform: translateY(30px);
  }

  to {
    opacity: 1;
    transform: translateY(0);
  }
`;
