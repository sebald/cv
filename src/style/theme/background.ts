import palette from './palette';

export default {
  body: palette.white,
};
