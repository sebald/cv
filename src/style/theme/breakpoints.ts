export default {
  small: 768,
  medium: 1024,
  large: 1280,
};
