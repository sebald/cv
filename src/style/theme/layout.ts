export default {
  sidebar: 350,
  content: 800,
};
