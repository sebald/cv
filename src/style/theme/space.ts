export default {
  baseline: 16,
};
