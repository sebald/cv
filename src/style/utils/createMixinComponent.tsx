import React from 'react';

import { styled } from '../styled-components';
import { StyledMixin } from '../theme';

import { mergeClassNames } from './classnames';
import { WithClassName } from './types';

/**
 * Idea stolen from:
 * https://github.com/jxnblk/grid-styled/tree/master/styled-space
 */
const StyledChildren: React.FC<WithClassName> = ({ className, children }) => {
  const styledChildren = React.Children.toArray(children).map((child) =>
    React.isValidElement<any>(child)
      ? React.cloneElement(
          child,
          mergeClassNames(child.props.className, className)
        )
      : child
  );

  return <>{styledChildren}</>;
};

export const createMixinComponent = <P extends {}>(
  ...mixins: StyledMixin<P>[]
) => styled(StyledChildren)([] as any, ...mixins);
