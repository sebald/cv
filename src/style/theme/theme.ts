import breakpoints from './breakpoints';
import layout from './layout';
import palette from './palette';
import space from './space';
import typography from './typography';

const theme = {
  breakpoints,
  layout,
  palette,
  space,
  typography,
};

export default theme;
