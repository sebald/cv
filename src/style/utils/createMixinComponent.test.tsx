import { mount } from 'enzyme';
import * as React from 'react';

import { StyledMixin } from '../theme';
import { createMixinComponent } from './createMixinComponent';

const mixinStatic = () => ({ fontSize: 'normal' });
const mixinDynamic: StyledMixin<{ width: number }> = ({ width }) => ({
  width: `${width}px`,
});

const Child: React.FC<{ className?: string }> = ({ className }) => (
  <div className={className} />
);

test('create a mixin component', () => {
  const Mixin = createMixinComponent(mixinStatic);
  const wrapper = mount(
    <Mixin>
      <Child />
    </Mixin>
  );

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find(Child)).toHaveStyleRule('font-size', 'normal');
});

test('merge classnames', () => {
  const Mixin = createMixinComponent(mixinStatic);
  const wrapper = mount(
    <Mixin className="custom-name">
      <Child />
    </Mixin>
  );

  expect(wrapper).toMatchSnapshot();
});

test('create a dynamic mixin component', () => {
  const Mixin = createMixinComponent(mixinDynamic);
  const wrapper = mount(
    <Mixin width={100}>
      <Child />
    </Mixin>
  );

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find(Child)).toHaveStyleRule('width', '100px');
});

test('multiple mixins', () => {
  const Mixin = createMixinComponent(mixinStatic, mixinDynamic);
  const wrapper = mount(
    <Mixin width={55}>
      <Child />
    </Mixin>
  );

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find(Child)).toHaveStyleRule('font-size', 'normal');
  expect(wrapper.find(Child)).toHaveStyleRule('width', '55px');
});

test('apply to all children', () => {
  const Mixin = createMixinComponent(mixinStatic);
  const wrapper = mount(
    <Mixin>
      <Child />
      <div />
      <span />
    </Mixin>
  );
  expect(wrapper).toMatchSnapshot();
});
