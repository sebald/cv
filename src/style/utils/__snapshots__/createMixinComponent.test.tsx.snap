// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apply to all children 1`] = `
.c0 {
  font-size: normal;
}

<Styled(StyledChildren)>
  <StyledChildren
    className="c0"
  >
    <Child
      className="c0"
      key=".0"
    >
      <div
        className="c0"
      />
    </Child>
    <div
      className="c0"
      key=".1"
    />
    <span
      className="c0"
      key=".2"
    />
  </StyledChildren>
</Styled(StyledChildren)>
`;

exports[`create a dynamic mixin component 1`] = `
.c0 {
  width: 100px;
}

<Styled(StyledChildren)
  width={100}
>
  <StyledChildren
    className="c0"
    width={100}
  >
    <Child
      className="c0"
      key=".0"
    >
      <div
        className="c0"
      />
    </Child>
  </StyledChildren>
</Styled(StyledChildren)>
`;

exports[`create a mixin component 1`] = `
.c0 {
  font-size: normal;
}

<Styled(StyledChildren)>
  <StyledChildren
    className="c0"
  >
    <Child
      className="c0"
      key=".0"
    >
      <div
        className="c0"
      />
    </Child>
  </StyledChildren>
</Styled(StyledChildren)>
`;

exports[`merge classnames 1`] = `
.c0 {
  font-size: normal;
}

<Styled(StyledChildren)
  className="custom-name"
>
  <StyledChildren
    className="custom-name c0"
  >
    <Child
      className="custom-name c0"
      key=".0"
    >
      <div
        className="custom-name c0"
      />
    </Child>
  </StyledChildren>
</Styled(StyledChildren)>
`;

exports[`multiple mixins 1`] = `
.c0 {
  font-size: normal;
  width: 55px;
}

<Styled(StyledChildren)
  width={55}
>
  <StyledChildren
    className="c0"
    width={55}
  >
    <Child
      className="c0"
      key=".0"
    >
      <div
        className="c0"
      />
    </Child>
  </StyledChildren>
</Styled(StyledChildren)>
`;
